src/Tribe/Attendee_Registration/Service_Provider.php

```diff
@@ -8,36 +8,112 @@ class Tribe__Tickets__Attendee_Registration__Service_Provider extends tad_DI52_S
+		$this->container->singleton( 'tickets.attendee_registration.view', Tribe__Tickets__Attendee_Registration__View::class );
+		$this->container->singleton( 'tickets.attendee_registration.rewrite', Tribe__Tickets__Attendee_Registration__Rewrite::class );
+		$this->container->singleton( 'tickets.attendee_registration.meta', Tribe__Tickets__Attendee_Registration__Meta::class );
+		$this->container->singleton( 'tickets.attendee_registration.shortcode', Tribe__Tickets__Attendee_Registration__Shortcode::class );
+		$this->container->singleton( 'tickets.attendee_registration.modal', Tribe__Tickets__Attendee_Registration__Modal::class );
+
+		// Priority 45 to be before priority 50 used on the same action in hooks().
+		add_action( 'tribe_plugins_loaded', [ $this, 'hooks' ], 45 );
+	}
+
+	/**
+	 * Add actions and filters for the Attendee Info classes.
+	 *
+	 * @since 4.9
+	 * @since TBD Bail if ETP is not active.
+	 */
+	public function hooks() {
+		// Only run the code if Event Tickets Plus is active.
 		if ( ! class_exists( 'Tribe__Tickets_Plus__Main' ) ) {
 			return;
 		}
 
-		tribe_singleton( 'tickets.attendee_registration', 'Tribe__Tickets__Attendee_Registration__Main' );
-		tribe_singleton( 'tickets.attendee_registration.template', 'Tribe__Tickets__Attendee_Registration__Template' );
-		tribe_singleton( 'tickets.attendee_registration.view', 'Tribe__Tickets__Attendee_Registration__View' );
-		tribe_singleton( 'tickets.attendee_registration.rewrite', 'Tribe__Tickets__Attendee_Registration__Rewrite' );
-		tribe_singleton( 'tickets.attendee_registration.meta', 'Tribe__Tickets__Attendee_Registration__Meta' );
-		tribe_singleton( 'tickets.attendee_registration.shortcode', 'Tribe__Tickets__Attendee_Registration__Shortcode' );
-		tribe_singleton( 'tickets.attendee_registration.modal', 'Tribe__Tickets__Attendee_Registration__Modal' );
+		// After priority 45 from register() using this same action.
+		add_action( 'tribe_plugins_loaded', [ $this, 'add_attendee_registration_template_hook' ], 50 );
+
+		add_action( 'init', [ $this, 'add_attendee_registration_shortcode_hook' ] );
+		add_action( 'init', [ $this, 'add_attendee_registration_modal_hook' ] );
+		add_action( 'init', [ $this, 'add_rewrite_tags' ] );
+		add_action( 'tribe_tickets_pre_rewrite', [ $this, 'generate_core_rules' ] );
+
+		add_filter( 'generate_rewrite_rules', [ $this, 'filter_generate' ] );
+		add_filter( 'rewrite_rules_array', [ $this, 'remove_percent_placeholders' ], 25 );
+		add_filter( 'tribe_tickets_commerce_paypal_add_to_cart_args', [ $this, 'add_product_delete_to_paypal_url' ], 10, 1 );
+	}
+
+	public function add_attendee_registration_template_hook() {
+		/** @var Tribe__Tickets__Attendee_Registration__Template $make */
+		$make = $this->container->make( 'tickets.attendee_registration.template' );
+
+		$make->hook();
+	}
+
+	public function add_attendee_registration_shortcode_hook() {
```

Comment:
:no_entry_sign: Missing doc comment for function add_attendee_registration_shortcode_hook()

<sub>hash: 7ce174cea6989cb62de016724fc1726a</sub>

Reply:
Needed